src/components/Labelling/LabellingFlow.js

```diff
@@ -62,23 +62,41 @@ export default class LabellingFlow extends Component {
 
     const style = Object.assign({}, this.state.componentStyle);
     if (this.state.skipAddLabelButton) {
-      style.left -= 160;
+      if (style.left - 160 < 0) {
+        style.left = 0;
+      } else {
+        style.left -= 160;
+      }
+    }
+
+    if (this.state.editLocation) {
+      style.maxHeight = '70vh';
+      if (!this.initialTopDistance) {
+        this.initialTopDistance = window.innerHeight - window.innerHeight * 0.3;
+        style.top = `${this.state.componentStyle.top -
+          this.initialTopDistance / 2}px`;
+      } else {
+        style.top = `${this.state.componentStyle.top}px`;
```

Comment:
We should be using CSS transforms rather than top/left so we can leverage the GPU.

https://www.html5rocks.com/en/tutorials/speed/high-performance-animations/